### PR TITLE
Add runOnlyAtRoot option

### PR DESCRIPTION
**Pull request by leonard84**
As requested by @longtimeago moved the option to a different PR #127

### CODE REVIEW

**Comment by leonard84** [comment]
The oraclejdk8 is not available anymore, so that build is failing. OpenJDK passes

**Comment by leonard84** [comment]
@longtimeago can you take a look, and maybe retrigger/fix the oracle build.

**Comment by longtimeago** [comment]
@leonard84 thanks a lot! will try to go through today

**Comment by longtimeago** [comment]
@leonard84 oraclejdk8 build failed again after re-trigger
```bash
Installing oraclejdk8
$ export JAVA_HOME=~/oraclejdk8
$ export PATH="$JAVA_HOME/bin:$PATH"
$ ~/bin/install-jdk.sh --target "/home/travis/oraclejdk8" --workspace "/home/travis/.cache/install-jdk" --feature "8" --license "BCL"
Ignoring license option: BCL -- using GPLv2+CE by default
install-jdk.sh 2019-07-17
Expected feature release number in range of 9 to 14, but got: 8
The command "~/bin/install-jdk.sh --target "/home/travis/oraclejdk8" --workspace "/home/travis/.cache/install-jdk" --feature "8" --license "BCL"" failed and exited with 3 during .
```

**Comment by leonard84** [comment]
@longtimeago if you fix it on master I can rebase, the problem is that oracle does not provide Java 8 anymore. My suggestion is to just remove it, you are already building on openjdk.
It doesn't have anything to do with the PR itself.

**Comment by longtimeago** [comment]
> My suggestion is to just remove it, you are already building on openjdk.

@mykelalvis would you agree to remove oracle build from travis?

**Comment by leonard84** [comment]
@longtimeago hi, I would really like a release with this feature, since it is the only plugin that is currently preventing us from using multi threaded maven.

**Comment by longtimeago** [comment]
@leonard84 sorry, man, I'm not a gatekeeper here. We should wait for @mickaelistria

**Comment by mickaelistria** [comment]
As already mentioned someplace else, I'm not able to nor much in need of remain/ing involved in this project at the moment.
@longtimeago I'll make you a owner of everything on GitHub so the project can move forward without me.

**Comment by mickaelistria** [comment]
@longtimeago what's you Sonatype OSSHR account id? I'll open a ticket to make you owner of the download-maven-plugin group there as well.

**Comment by longtimeago** [comment]
@mickaelistria alright .. I wasn't aware that you want to transfer an ownership entirely, but sure, why not :)
My Sonatype OSSHR account id: `longtimeago`

**Comment by mickaelistria** [comment]
> alright .. I wasn't aware that you want to transfer an ownership entirely, but sure, why not :)

I don't think keeping me as a bottleneck here and there is good for the project, so I'm glad to see someone else taking care of it and serve it better than I have done lately and am likely to be in near future ;)
FYI, last release was performed by @adamretter IIRC, with documentation at https://github.com/maven-download-plugin/maven-download-plugin/wiki , so maybe you 2 can collaborate on releasing the bits. You can always ping me if you need some assistance or some more permissions I missed to grant you.

> My Sonatype OSSHR account id: longtimeago

I opened https://issues.sonatype.org/browse/OSSRH-51261. I couldn't find a way to link to your account so I think you need to manually subscribe to this issue to get notifications.

**Comment by longtimeago** [comment]
@mickaelistria thank you for the trust :)

> FYI, last release was performed by

@mickaelistria I'll try alone first and ping you or Adam in case of a problem. ;)
In general I have idea to automate the release process, but definitely not now ...

**Comment by longtimeago** [comment]
@leonard84 I've removed oracle jdk from travis. Please, rebase the PR :)

**Comment by leonard84** [comment]
@longtimeago done

File: src/main/java/com/googlecode/download/maven/plugin/internal/WGet.java [hunk 1]
```diff
@@ -73,7 +74,7 @@
  * @author Marc-Andre Houle
  * @author Mickael Istria (Red Hat Inc)
  */
-@Mojo(name = "wget", defaultPhase = LifecyclePhase.PROCESS_RESOURCES, requiresProject = false)
+@Mojo(name = "wget", defaultPhase = LifecyclePhase.PROCESS_RESOURCES, requiresProject = true)
```
Comment (longtimeago) [inline src/main/java/com/googlecode/download/maven/plugin/internal/WGet.java]: @leonard84 sorry, missed this thing at the initial CR.
Could we avoid requiring a project?

Comment (leonard84) [reply src/main/java/com/googlecode/download/maven/plugin/internal/WGet.java]: Why? and not really `if (runOnlyAtRoot && !project.isExecutionRoot())`

Comment (longtimeago) [reply src/main/java/com/googlecode/download/maven/plugin/internal/WGet.java]: @leonard84 uff, it's already set to `true` by default (https://maven.apache.org/plugin-tools/maven-plugin-annotations/apidocs/org/apache/maven/plugins/annotations/Mojo.html#requiresProject--) so you just made it explicit. Nothing to worry about

**Comment by longtimeago** [comment]
@leonard84 would the release at of this particular step is valuable for you or you wanna go with the second tricky part (out of your initial PR)?

**Comment by leonard84** [comment]
@longtimeago yes this would actually solve our problem, since we only download one file for a project.

**Pull request merged**